pyigm/cgm/cos_halos.py

```diff
@@ -0,0 +1,423 @@
+        hdu = fits.open(fil)
+        summ = Table(hdu[1].data)
+        galx = Table(hdu[2].data)
+        # Instantiate the galaxy
+        gal = Galaxy((galx['RA'][0], galx['DEC'][0]), z=summ['ZFINAL'][0])
+        gal.field = galx['FIELD'][0]
+        gal.gal_id = galx['GALID'][0]
+        # Galaxy properties
+        gal.halo_mass = summ['LOGMHALO'][0]
+        gal.stellar_mass = summ['LOGMFINAL'][0]
+        gal.sfr = (galx['SFR_UPLIM'][0], galx['SFR'][0],
+                                       galx['SFR_FLAG'][0]) # FLAG actually gives method used
+        # Instantiate the IGM System
+        igm_sys = IGMSystem('CGM',(galx['QSORA'][0], galx['QSODEC'][0]),
+                            summ['ZFINAL'][0], [-400, 400.]*u.km/u.s)
+        igm_sys.zqso = galx['ZQSO'][0]
+        # Metallicity
+        igm_sys.ZH = -99.
+        if self.cldy is not None:
+            mtc = np.where((self.cldy['GALID'] == gal.gal_id) &
+                           (self.cldy['FIELD'] == gal.field))[0]
+            if len(mtc) == 1:
+                igm_sys.ZH = self.cldy[mtc]['ZBEST'][0]
+        # Instantiate
+        self.cgm_abs.append(CGMAbsSys(gal, igm_sys))
+        # Ions
+        if skip_ions is True:
+            return
+        mm = len(self.cgm_abs)-1
+        all_Z = []
+        all_ion = []
+        for jj in range(summ['NION'][0]):
+            iont = hdu[3+jj].data
+            if jj == 0: # Generate new Table
+                dat_tab = Table(iont)
+            else:
+                try:
+                    dat_tab.add_row(Table(iont)[0])
+                except:
+                    pdb.set_trace()
+            all_Z.append(iont['ZION'][0][0])
+            all_ion.append(iont['ZION'][0][1])
+        # Add Z,ion
+        dat_tab.add_column(Column(all_Z,name='Z'))
+        dat_tab.add_column(Column(all_ion,name='ion'))
+        # Rename
+        dat_tab.rename_column('LOGN','indiv_logN')
+        dat_tab.rename_column('SIGLOGN','indiv_sig_logN')
+        dat_tab.rename_column('CLM','logN')
+        dat_tab.rename_column('SIG_CLM','sig_logN')
+        dat_tab.rename_column('FLG_CLM','flag_N')
+        # Set
+        self.cgm_abs[mm].igm_sys._ionN = dat_tab
+        # NHI
+        self.cgm_abs[mm].igm_sys.NHI = dat_tab[
+            (dat_tab['Z']==1)&(dat_tab['ion']==1)]['logN'][0]
+        self.cgm_abs[mm].igm_sys.flag_NHI = dat_tab[
+            (dat_tab['Z']==1)&(dat_tab['ion']==1)]['flag_N'][0]
+
+    # Load from mega structure
+    def load_mega(self,data_file=None, cosh_dct=None, test=False, **kwargs):
+        """ Load the data for COS-Halos
+
+        Paramaeters
+        ----------
+        data_file : string
+          Name of data file
+        pckl_fil : string
+          Name of file for pickling
+        """
+        # Loop
+        if test is True:
+            cos_files = glob.glob(self.fits_path+'/J091*.fits.gz')  # For testing
+        else:
+            cos_files = glob.glob(self.fits_path+'/J*.fits.gz')
```

Comment:
If I'm following this correctly, those fits.gz files should be within something like "~/Dropbox/COS-Halos/Summary/", but I don't see them in my copy of the Dropbox?  For the COS-Dwarfs data, I converted the IDL .sav files to .fits, and all the data were in 'cosmetals_megastructure.new.sav'.   Is there a need to loop through the structures for each COS-Halos galaxy?